src/util.c

```diff
@@ -656,11 +715,20 @@ void dcc_get_proc_stats(int *num_D, int *max_RSS, char **max_RSS_name) {
     }
 
     *num_D = 0;
+    *mem_available = -1;
     *max_RSS = 0;
     *max_RSS_name = RSS_name;
     RSS_name[0] = 0;
 
     while ((procsubdir = readdir(proc)) != NULL) {
+        if (!checked_meminfo &&
+                strncmp(procsubdir->d_name, "meminfo", sizeof("meminfo"))) {
+            f = fopen("/proc/meminfo", "r");
```

Comment:
Not a big deal, but I think this doesn't really need to be inside the loop? You could just attempt to read it before or after the loop, and if for some reason it's not readable you'll get -1.